bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/DateTime.java

```diff
@@ -1868,6 +1875,38 @@ boolean onNumberKeyInput(int key) {
 	return false;
 }
 
+private StringBuilder validateNewTimeValue(String prefix, String keyValue) {
+	if(typeBuffer.toString().isEmpty()) return typeBuffer;
+	int numColons = (prefix.length() - prefix.replace(":", "").length()) / ":".length();
```

Comment:
Not all locales use ":" as time separator.
Some use "." (be (Belarusian), it (Italian), lt (Lithuanian), sq (Albanian), sr (Serbian))

Worse, this is not 100% stable, might change between Java versions.
The list above is from Java 8.
Java 17 moved to use CLDR by default, so the list of locales using "." grows (with Assamaese, Danish, more)
Worse, separators in 17 go beyond ":" and ".", French-Canada uses " h " ("19 h 43").
And 19:43 in Dzonga (dz) is "ཆུ་ཚོད་ ༧ སྐར་མ་ ༤༣ ཕྱི་ཆ་" (the separator is "སྐར་མ་")

Reply:
Updated PR to better handle different locales.